noxfile.py

```diff
@@ -35,3 +36,69 @@ def mypy(session: nox.Session):
 def test(session: nox.Session):
     session.install("pytest", ".")
     session.run("pytest", "setuptools_rust", *session.posargs)
+
+
+def download_pyodide(session: nox.Session, pyodide_dir: Path) -> None:
+    pyodide_dir.mkdir()
+
+    PYODIDE_DEV = "https://pyodide-cdn2.iodide.io/dev/full/"
+    pyodide_files = [
+        "pyodide.js",
+        "packages.json",
+        "pyodide.asm.js",
+        "pyodide.asm.data",
+        "pyodide.asm.wasm",
+        "pyodide_py.tar",
+    ]
+    with session.chdir(pyodide_dir):
+        for file in pyodide_files:
+            session.run("wget", PYODIDE_DEV + file, external=True)
+        session.run("npm", "i", "node-fetch", external=True)
+
+
+@nox.session(name="test-examples-emscripten")
+def test_examples_emscripten(session: nox.Session):
+    session.install(".")
+    emscripten_dir = Path("./emscripten").resolve()
+    pyodide_dir = emscripten_dir / "pyodide"
+    if not pyodide_dir.exists():
+        download_pyodide(session, pyodide_dir)
+
+    session.run(
+        "rustup",
+        "component",
+        "add",
+        "rust-src",
+        "--toolchain",
+        "nightly",
+        external=True,
+    )
+    for example in Path("examples/").resolve().glob("*"):
+        if example.name == "hello-world":
+            continue
+        env = os.environ.copy()
+        env.update(
+            RUSTUP_TOOLCHAIN="nightly",
+            _SETUPTOOLSRUST_BUILD_STD="1",
+            PYTHONPATH=str(emscripten_dir),
+            _PYTHON_SYSCONFIGDATA_NAME="_sysconfigdata__emscripten_wasm32-emscripten",
+            _PYTHON_HOST_PLATFORM="emscripten_3_1_13_wasm32",
+            CARGO_BUILD_TARGET="wasm32-unknown-emscripten",
+            CARGO_TARGET_WASM32_UNKNOWN_EMSCRIPTEN_LINKER=str(
+                emscripten_dir / "emcc_wrapper.py"
+            ),
+            PYO3_CONFIG_FILE=str(emscripten_dir / "pyo3_config.ini"),
```

Comment:
From this should I infer that `pyo3_build_config` doesn't configure for pyodide correctly? Do you think we would be able to upstream a fix?

Reply:
> From this should I infer that pyo3_build_config doesn't configure for pyodide correctly?

I am not sure. I think it may just be that I don't know how to use the tool correctly. How would I check this?

Reply:
So usually in cross-compile situations like for emscripten the `pyo3_build_config` build scripts will try to look for a sysconfigdata file and configure from that. Alternatively, can set the `abi3-py310` feature and then it will do a "standard" cross compile, which might just work.

I view the `PYO3_CONFIG_FILE` env var as an escape hatch when nothing built-in works, so if either of the above work (or can be made to work) that's probably preferable.